compiler/rustc_trait_selection/src/opaque_types.rs

```diff
@@ -76,7 +82,69 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         ));
         debug!(?definition_ty);
 
-        definition_ty
+        // Only check this for TAIT. RPIT already supports /home/ubuntu/rust8/src/test/ui/impl-trait/nested-return-type2.rs
```

Comment:
```suggestion
        // Only check this for TAIT. RPIT already supports `src/test/ui/impl-trait/nested-return-type2.rs`
```

I don't think the compiler source needs to record that this was fixed in `rust8` for the rest of time :)